server/src/main/java/org/apache/druid/server/SegmentManager.java

```diff
@@ -192,12 +194,14 @@ public boolean loadSegment(final DataSegment segment) throws SegmentLoadingExcep
   private Segment getAdapter(final DataSegment segment) throws SegmentLoadingException
   {
     final Segment adapter;
-    try {
-      adapter = segmentLoader.getSegment(segment);
-    }
-    catch (SegmentLoadingException e) {
-      segmentLoader.cleanup(segment);
-      throw e;
+    synchronized (segmentLoader) {
```

Comment:
This is too much synchronization - it makes parallel segment loading impossible.

Reply:
Oh right. Thank you for catching. I'll fix this.